### ISSUE TITLE
Nullable<T>.GetType() should warn about possible null values and update the accessed variable's nullablility

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#33174]
**Version Used**: 
Microsoft Visual Studio Enterprise 2019 Preview
Version 16.0.0 Preview 2.1
VisualStudio.16.Preview/16.0.0-pre.2.1+28529.54
Microsoft .NET Framework
Version 4.7.03056

**Steps to Reproduce**:

Compile and run the following code:
```cs
#nullable enable
class C
{
  static void Main()
  {
    int? x = null;
    x.GetType(); // missing warning
    var y = (int) x; // incorrect warning
  }
}
```
**Expected Behavior**:
Warning `CS8629 Nullable value type may be null.` for `x.GetType()`.
No warnings for the line `var y = (int) x;` since x should be known to be not null after `x.GetType()`

**Actual Behavior**:
No warnings for `x.GetType()`. The program crashes at runtime at this line with a `NullReferenceException`. Incorrect warning CS8629 for `var y = (int) x;`.
